src/dictionaries.cpp

```diff
@@ -85,21 +103,28 @@ Dictionaries::Dictionaries(const std::string& filePath)
                 vendorName = tokens[1];
             else if (tokens[0] == "END-VENDOR")
                 vendorName.clear();
+            else if (tokens[0] == "$INCLUDE")
+            {
+                if (tokens[1].substr(0, 1) == "/")
+                {
+                    Dictionaries fillingDictionaries(tokens[1]);
+                    m_attributes.append(fillingDictionaries.m_attributes);
+                    m_vendorNames.append(fillingDictionaries.m_vendorNames);
+                    m_attributeValues.append(fillingDictionaries.m_attributeValues);
+                    m_vendorAttributes.append(fillingDictionaries.m_vendorAttributes);
+                    m_vendorAttributeValues.append(fillingDictionaries.m_vendorAttributeValues);
+                }
+                else
+                {
+                    Dictionaries fillingDictionaries(filePath.substr(0, filePath.rfind('/') + 1) + tokens[1]);
+                    m_attributes.append(fillingDictionaries.m_attributes);
+                    m_vendorNames.append(fillingDictionaries.m_vendorNames);
+                    m_attributeValues.append(fillingDictionaries.m_attributeValues);
+                    m_vendorAttributes.append(fillingDictionaries.m_vendorAttributes);
+                    m_vendorAttributeValues.append(fillingDictionaries.m_vendorAttributeValues);
```

Comment:
I think it makes sense to add `append` method to the `Dictionaries` class. It will contain all these `append` calls for its members. And after adding the code above can be written as:
```c++
if (tokens[1].substr(0, 1) == "/")
    append(Dictionaries(tokens[1]));
else
    append(Dictionaries(filePath.substr(0, filePath.rfind('/') + 1) + tokens[1]));
```